src/Controller/Api/AbstractController.php

```diff
@@ -34,67 +34,54 @@
 
 class AbstractController extends \Symfony\Bundle\FrameworkBundle\Controller\AbstractController
 {
-    /**
-     * @var \JMS\Serializer\Serializer
-     */
-    protected $serializer;
+    protected SerializerInterface $serializer;
 
     public function __construct(SerializerInterface $serializer)
     {
         $this->serializer = $serializer;
     }
 
-    /**
-     * @param string $version
-     * @return \App\Entity\MajorVersion
-     */
     protected function findMajorVersion(string $version): MajorVersion
     {
         $this->checkMajorVersionFormat($version);
-        $repo = $this->getDoctrine()->getRepository(MajorVersion::class);
-        $entity = $repo->findOneBy(['version' => $version]);
-        if (null === $entity) {
+        $majorVersion = $this->getDoctrine()->getRepository(MajorVersion::class)->findOneBy(['version' => $version]);
+        if (!$majorVersion instanceof MajorVersion) {
             throw new NotFoundHttpException('No such version.');
         }
-        return $entity;
+        return $majorVersion;
     }
 
     /**
-     * @param \Symfony\Component\Validator\Validator\ValidatorInterface $validator
      * @param $object
      */
     protected function validateObject(ValidatorInterface $validator, $object): void
     {
-        $errors = $validator->validate($object);
+        $constraintViolationList = $validator->validate($object);
```

Comment:
Maybe just `$violations`?  Still descriptive, still clear that it's a list of violations, but easier to read and type.

Reply:
Yes, agree.